src/cogserver/vrt.py

```diff
@@ -0,0 +1,99 @@
+from typing import List, Literal, Optional
+from enum import Enum
+from fastapi import APIRouter, Query, Response
+from titiler.core.factory import FactoryExtension, TilerFactory, BaseTilerFactory, MultiBaseTilerFactory
+from osgeo import gdal
+import tempfile
+
+router = APIRouter()
+
+
+class ResampleAlg(Enum):
+    nearest: str = "nearest"
+    bilinear: str = "bilinear"
+    cubic: str = "cubic"
+    cubicspline: str = "cubicspline"
+    lanczos: str = "lanczos"
+    average: str = "average"
+    mode: str = "mode"
+
+
+async def create_vrt_from_urls(
+        urls: List[str],
+        xRes: float = 0.1,
+        yRes: float = 0.1,
+        vrtNoData: List[str] = 0,
+        srcNoData: List[str] = 0,
+        resamplingAlg: Literal["nearest", "bilinear", "cubic", "cubicspline", "lanczos", "average", "mode"] = "nearest",
+        resolution: Literal["highest", "lowest", "average", "user"] = "average"
+):
+    urls = [f"/vsicurl/{url}" for url in urls]
+    if vrtNoData:
+        vrtNoData = " ".join(vrtNoData)
+    if srcNoData:
+        srcNoData = " ".join(srcNoData)
+
+    options = gdal.BuildVRTOptions(
+        separate=True,
+        bandList=list(range(1, len(urls) + 1)),
+        xRes=xRes,
+        yRes=yRes,
+        resampleAlg=resamplingAlg,
+        VRTNodata=vrtNoData,
+        srcNodata=srcNoData,
+        resolution=resolution
+    )
+
+    with tempfile.NamedTemporaryFile() as temp:
+        gdal.BuildVRT(temp.name, urls, options=options)
+        with open(temp.name, "r") as file:
+            return file.read()
+
+
+class VRTFactory(MultiBaseTilerFactory):
+    """
+    Override the MultiBaseTilerFactory to add a VRT endpoint
+    Empty register_routes method to override all routes and have no routes
+    """
+
+    def register_routes(self):
+        pass
+
+
+class VRTExtension(FactoryExtension):
+    def register(self, factory: VRTFactory):
+        @factory.router.get(
+            "/",
+            response_class=Response,
+            responses={200: {"description": "Return a VRT from multiple COGs."}},
+        )
+        async def create_vrt(
+                urls: List[str] = Query(..., description="Dataset URLs"),
+
```

Comment:
Could you move `VRTExtension` class to different file?

There is already `extensions.py`, we could move VRTExtension to this python file.

But I don't actually like adding many extension class in a python file, we could introduce `extensions` folder as below folder structure

- `extensions` folder
  - `__init__.py`
  - mosaicjson.py: `createMosaicJsonExtension` class (I think we can rename it to just `MosaicJsonExtension` class by removing verb of `create`.
  - vrt.py: `VRTExtension` class